### DISCUSSION TITLE
Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)

### DISCUSSION THREAD

**Discussion by ouening** [alandefreitas/matplotplusplus#67] [Q&A]
Hello, I'm using mingw64 7.3 in Win10, when I mkdir build && cd build && cmake .. -G "MinGW Makefiles" , I got errors like title said. How can I solve it?

**Comment by alandefreitas** [comment]
Hi @ouening,
Can you share the steps to reproduce the problem?
What errors did you get? What's the exact output?

Maybe you can look at the continuous integration and compare it with what you are doing to see where things are going differently:
https://github.com/alandefreitas/matplotplusplus/actions/runs/277860875
https://github.com/alandefreitas/matplotplusplus/runs/1180165617?check_suite_focus=true

**Comment by ouening** [comment]
Steps to reproduce the problem:
1. Win10 system. Prepare mingw64 7.3 version, cmake , cmder (Linux like terminal), add them to system environment variable
2. Clone project and change directory to it
3. run: 
```
mkdir build
cd build 
cmake .. -G "MinGW Makefiles"
```
Then errors happened.

**Reply by alandefreitas** [reply]
What errors? Have you tried comparing with the [windows integration](https://github.com/alandefreitas/matplotplusplus/runs/1180165617?check_suite_focus=true) script?

**Reply by alandefreitas** [reply]
Is that the error

```
Could NOT find PkgConfig (missing:  PKG_CONFIG_EXECUTABLE)
```

in the middle of the build script?

Is it a warning or a fatal error? If it's not fatal, you can just ignore it. It must be something like [that](https://github.com/AcademySoftwareFoundation/openvdb/issues/571).

**Reply by alandefreitas** [reply]
In the windows integration script we have 

```
CMAKE_MODULE_PATH=D:/a/matplotplusplus/matplotplusplus/cmake
17
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
18
    Reason given by package: The command
19
      "C:/Strawberry/perl/bin/pkg-config" --version
20
    failed with output
21
      
22

23
-- Could NOT find JPEG (missing: JPEG_LIBRARY) (found version "90")
24
-- Could NOT find TIFF (missing: TIFF_LIBRARY) (found version "4.0.10")
25
-- Could NOT find ZLIB (missing: ZLIB_LIBRARY) (found version "1.2.11")
26
-- Looking for sgemm_
27
```

If that's the error, you can ignore it. It's just a regular message.

**Comment by ouening** [comment]
The error is:
```
λ cmake ..
CMAKE_MODULE_PATH=D:/GoogleDownload/matplotplusplus-master/cmake
-- Could NOT find JPEG (missing: JPEG_LIBRARY JPEG_INCLUDE_DIR)
-- Could NOT find TIFF (missing: TIFF_LIBRARY TIFF_INCLUDE_DIR)
-- Could NOT find ZLIB (missing: ZLIB_INCLUDE_DIR)
-- A library with LAPACK API found.
CMake Error at cmake/FindFilesystem.cmake:246 (message):
  Cannot run simple program using std::filesystem
Call Stack (most recent call first):
  source/matplot/CMakeLists.txt:1 (find_package)


-- Configuring incomplete, errors occurred!
See also "D:/GoogleDownload/matplotplusplus-master/build/CMakeFiles/CMakeOutput.log".
See also "D:/GoogleDownload/matplotplusplus-master/build/CMakeFiles/CMakeError.log".
```
Sorry, I don't use MSVC in Win10，so I think windows integration script won't help.

**Reply by alandefreitas** [reply]
I see. The real error is not `missing: PKG_CONFIG_EXECUTABLE` then. The real error is:

```
CMake Error at cmake/FindFilesystem.cmake:246 (message):
  Cannot run simple program using std::filesystem
```

It means your compiler doesn't have the C++17 filesystem library. You need to update it.

**Comment by ouening** [comment]
Yeah, I will try new mingw
